apps/apache.go

```diff
@@ -0,0 +1,171 @@
+			"Nest_under":    "logging.googleapis.com/http_request",
+			"Remove_prefix": "http_request_",
+		},
+	})
+	return c
+}
+
+type LoggingProcessorApacheError struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorApacheError) Type() string {
+	return "apache_error"
+}
+
+func (p LoggingProcessorApacheError) Components(tag string, uid string) []fluentbit.Component {
+	c := confgenerator.LoggingProcessorParseRegex{
+		// Documentation: https://httpd.apache.org/docs/current/logs.html#errorlog
+		// Sample line 2.4: [Fri Sep 09 10:42:29.902022 2011] [core:error] [pid 35708:tid 4328636416] (13)Permission denied [client 72.15.99.187] File does not exist: /usr/local/apache2/htdocs/favicon.ico
+		// Sample line 2.2: [Fri Sep 09 10:42:29.902022 2011] [error] [pid 35708:tid 4328636416] [client 72.15.99.187] File does not exist: /usr/local/apache2/htdocs/favicon.ico
+		// TODO - Support time parsing for version 2.0 where smallest resolution is seconds
+		// Sample line 2.0: [Wed Oct 11 14:32:52 2000] [error] [client 127.0.0.1] client denied by server configuration: /export/home/live/ap/htdocs/test
+		Regex: `^\[(?<time>[^\]]+)\] \[(?:(?<module>\w+):)?(?<level>[\w\d]+)\](?: \[pid (?<pid>\d+)(?::tid (?<tid>[0-9]+))?\])?(?: (?<errorCode>[^\[]*))?(?: \[client (?<client>[^\]]*)\])? (?<message>.*)$`,
+		LoggingProcessorParseShared: confgenerator.LoggingProcessorParseShared{
+			TimeKey:    "time",
+			TimeFormat: "%a %b %d %H:%M:%S.%L %Y",
+			Types: map[string]string{
+				"pid": "integer",
+				"tid": "integer",
+			},
+		},
+	}.Components(tag, uid)
+	for _, l := range []struct{ level, severity string }{
+		// Log levels documented: https://httpd.apache.org/docs/2.4/mod/core.html#loglevel
+		// Could separate traceN out into a Key_Value_Matches modify condition but left it in
+		// the group for simplicity and clarity of the translations.
+		{"emerg", "EMERGENCY"},
+		{"alert", "ALERT"},
+		{"crit", "CRITICAL"},
+		{"error", "ERROR"},
+		{"warn", "WARNING"},
+		{"notice", "NOTICE"},
+		{"info", "INFO"},
+		{"debug", "DEBUG"},
+		{"trace1", "DEBUG"},
+		{"trace2", "DEBUG"},
+		{"trace3", "DEBUG"},
+		{"trace4", "DEBUG"},
+		{"trace5", "DEBUG"},
+		{"trace6", "DEBUG"},
+		{"trace7", "DEBUG"},
+		{"trace8", "DEBUG"},
+	} {
+		c = append(c, fluentbit.Component{
+			Kind: "FILTER",
+			Config: map[string]string{
+				"Name":      "modify",
+				"Match":     tag,
+				"Condition": fmt.Sprintf("Key_Value_Equals level %s", l.level),
+				"Add":       fmt.Sprintf("logging.googleapis.com/severity %s", l.severity),
+			},
+		})
+	}
+	return c
+}
+
+type LoggingReceiverApacheAccess struct {
+	LoggingProcessorApacheAccess            `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+func (r LoggingReceiverApacheAccess) Components(tag string) []fluentbit.Component {
+	if len(r.IncludePaths) == 0 {
+		r.IncludePaths = []string{"/var/log/apache2/access.log", "/var/log/httpd/access_log"}
```

Comment:
Did we confirm these paths cover all of the supported Ops Agent distros? (The former is definitely current Debian/Ubuntu, the latter I believe is current Fedora, what about SLES etc.?)

Reply:
Filed a bug yesterday to extend the 3rd party app automated tests to more distros, then found there is already a WIP ticket for it: b/198830714. It requires some additional work. For now, we might need to manually spot check SLES.

Reply:
According to a bit of searching it looks like the SLES paths are similar to Debain/Ubuntu but with and underscore rather than a .log ("/var/log/apache2/error_log" for example). My initial searching I misread them to be the same as D/U. Added the paths to the defaults.

Reply:
Looks like this has been updated. Resolving this thread.